crates/ika-core/src/dwallet_mpc/crytographic_computation/mpc_computations/presign.rs

```diff
@@ -4,66 +4,327 @@
+                let advance_request = mpc_computations::try_ready_to_advance::<
+                    PresignParty<Secp256K1TaprootProtocol>,
+                >(
+                    party_id,
+                    access_structure,
+                    consensus_round,
+                    &serialized_messages_by_consensus_round,
+                )?;
+
+                advance_request.map(PresignAdvanceRequestByProtocol::Taproot)
+            }
+            DWalletSignatureScheme::SchnorrkelSubstrate => {
+                let advance_request = mpc_computations::try_ready_to_advance::<
+                    PresignParty<RistrettoSchnorrkelSubstrateProtocol>,
+                >(
+                    party_id,
+                    access_structure,
+                    consensus_round,
+                    &serialized_messages_by_consensus_round,
+                )?;
+
+                advance_request.map(PresignAdvanceRequestByProtocol::SchnorrkelSubstrate)
             }
+            DWalletSignatureScheme::EdDSA => {
+                let advance_request = mpc_computations::try_ready_to_advance::<
+                    PresignParty<Curve25519EdDSAProtocol>,
+                >(
+                    party_id,
+                    access_structure,
+                    consensus_round,
+                    &serialized_messages_by_consensus_round,
+                )?;
+
+                advance_request.map(PresignAdvanceRequestByProtocol::EdDSA)
+            }
+            DWalletSignatureScheme::ECDSASecp256r1 => {
+                let advance_request =
+                    mpc_computations::try_ready_to_advance::<PresignParty<Secp256R1ECDSAProtocol>>(
+                        party_id,
+                        access_structure,
+                        consensus_round,
+                        &serialized_messages_by_consensus_round,
+                    )?;
+
+                advance_request.map(PresignAdvanceRequestByProtocol::Secp256r1ECDSA)
+            }
+        };
+        Ok(advance_request)
+    }
+}
+
+impl PresignPublicInputByProtocol {
+    pub(crate) fn try_new(
+        session_identifier: SessionIdentifier,
+        protocol: DWalletSignatureScheme,
+        versioned_network_encryption_key_public_data: &VersionedNetworkEncryptionKeyPublicData,
+        dwallet_public_output: Option<SerializedWrappedMPCPublicOutput>,
+        protocol_version: ProtocolVersion,
+    ) -> DwalletMPCResult<Self> {
+        let dkg_output = dwallet_public_output
+            .clone()
+            .ok_or(DwalletMPCError::MPCSessionError {
+                session_identifier,
+                error: "presign public input cannot be None as we only support ECDSA".to_string(),
+            })?;
+
+        match protocol_version.as_u64() {
+            1 => Self::try_new_v1(
+                session_identifier,
+                versioned_network_encryption_key_public_data,
+                dkg_output,
+            ),
+            _ => Self::try_new_v2(
```

Comment:
```suggestion
            2 => Self::try_new_v2(
```
and then `_ => error!()`